src/Planner/Planner.cpp

```diff
@@ -820,7 +820,7 @@ void addWindowSteps(QueryPlan & query_plan,
         }
 
         auto window_step
-            = std::make_unique<WindowStep>(query_plan.getCurrentDataStream(), window_description, window_description.window_functions);
+            = std::make_unique<WindowStep>(query_plan.getCurrentDataStream(), window_description, window_description.window_functions, settings.query_plan_preserve_num_streams_after_window_functions);
```

Comment:
why we don't check `(i + 1) == windows_sorted.size()` here?

Reply:
fixed in latest commit